docs/design/design-patterns/non-functional-requirements_investigation_guide.md

```diff
@@ -0,0 +1,148 @@
+# Non-Functional Requirements Investigation
+
+## Background
+
+In software engineering, detailed investigation of non-functional requirements is often deferred until later stages of the project in favor of a more exclusive initial focus on the functional requirements.
+
+However, introducing non-functional requirements in the later stages of the SDLC is often highly-disruptive and inefficient to the ongoing engineering process.  Because of this, important characteristics of the system designed to provide for necessary e.g., testability, reliability, scalability, observability, securability, manageability are better considered as first-class citizens in the requirements gathering process to be defined in detail early in the engagement.
```

Comment:
nit: I prefer sentences on newlines in Markdown.

Reply:
Please expand the acronym "SDLC" before using it for the first time. I assume you mean "Software Development Life Cycle"

Reply:
Can you elaborate on the comment re: newlines for sentences?  Do you literally mean _each_ sentence should be on a separate line?  Since its a non-trivial amount of editing, just want to ensure correct understanding before I proceed to make the changes.  So this preceding para in this reply would instead be....

```
Can you elaborate on the comment re: newlines for sentences?
Do you literally mean _each_ sentence should be on a separate line?
Since its a non-trivial amount of editing, just want to ensure correct understanding before I proceed to make the changes.  So this preceding para in this reply would instead be....
```

Reply:
(also, SDLC initialism explained in first-use)

Reply:
Added newlines for each sentence per request.  With reduced verbiage, no longer a significant edit task :)